pom.xml

```diff
@@ -142,6 +124,27 @@
       <version>0.16</version>
       <scope>test</scope>
     </dependency>
+    <!-- Since 2.17, started using Mockito instead of Powermock -->
+    <!-- For testing with static methods -->
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>4.11.0</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-inline</artifactId>
+      <version>4.11.0</version>
+      <scope>test</scope>
+    </dependency>
+    <!-- We need to specify later Bytebuddy version because Mockito 4.11.0 depends on earlier Bytebuddy which
+    does not support JDK 21, since 2.17-->
+    <dependency>
+      <groupId>net.bytebuddy</groupId>
+      <artifactId>byte-buddy</artifactId>
+      <version>1.14.9</version>
+    </dependency>
```

Comment:
Phew, took a while to fix the JDK 21 failure. The problem was that

- We need Mockito 4.x because Mockito 5.x baseline jdk is 11 (where as we support jdk 8)
- Mockito 4.x depends on earlier ByteBuddy  version that does not support JDK 21

Reply:
Fun fun. Yeah, things will become more challenging going forward, no doubt.

Reply:
@JooHyukKim @cowtowncoder any reason for ByteBuddy not being scoped to `test`?

Reply:
Oh F*CK. This is bad. How did we miss this. :-(

Reply:
just release note it, no need to release 2.17.1 immediately - it is easy to work around (by excluding it in your build) and most people won't even notice the extra jar - needs to be fixed but not something to worry too much about

Reply:
Yes, this need not force release of 2.17.1